pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"time"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(fn func() error) error {
+	// Run query immediately
+	if err := fn(); !isTransientError(err) {
+		return err
+	}
+
+	expiration := time.NewTimer(timeout)
+	defer expiration.Stop()
```

Comment:
If you don't use a context, use 
```suggestion
	defer timeutil.StopTimer(expiration)
```
to prevent a resource leak